server.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2018 Artificial Solutions. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const dotenv = require('dotenv');
+dotenv.config();
+
+const { BasicCard, actionssdk } = require('actions-on-google');
+const express = require('express');
+const bodyParser = require('body-parser');
+const TIE = require('@artificialsolutions/tie-api-client');
+
+const teneoApi = TIE.init(process.env.TENEO_ENGINE_URL);
+
+const assistant = actionssdk();
+
+// initialise session handler, to store mapping between conversationId and engine session id
+const sessionHandler = SessionHandler();
+
+// Hande main intent
+assistant.intent('actions.intent.MAIN', async conv => {
+
+    // send message to teneo and return response
+    await handleMessage(conv);
+
+})
+
+// handle text inputs
+assistant.intent('actions.intent.TEXT', async conv => {
+
+    // send message to teneo and return response
+    await handleMessage(conv);
+
+})
+
+async function handleMessage(conv) {
+
+    // get the id of the conversation so we can map it to an engine session
+    var userToken = conv.id;
```

Comment:
In modern JavaScript, it's recommended to use `let` and `const` instead of `var` for variable declaration and it's also more consistent (for example, below in message `let` it's correctly used, not `var`)